lib/liquid/tokenizer.rb

```diff
@@ -5,7 +5,7 @@ class Tokenizer
     attr_reader :line_number, :for_liquid_tag
 
     def initialize(source, line_numbers = false, line_number: nil, for_liquid_tag: false)
-      @source         = source.to_s
+      @source         = source.to_s.to_str
```

Comment:
For posterity: 

This is necessary to handle cases where `to_s` returns a subclass of String but we explicitly want to turn it into an instance of String. As stated in this PR, `ActiveSupport::SafeBuffer` is an subclass of string, calling `ActiveSupport::SafeBuffer#to_s` returns self (an instance of safe_buffer) but we actually want an instance of string. Calling `to_str` ensures we're always dealing with an instance of String

Reply:
`to_str` is only defined on String: https://docs.ruby-lang.org/en/3.3/String.html#method-i-to_str